CONTRIBUTING.adoc

```diff
@@ -54,194 +66,225 @@ However, to avoid doing too much work all the time, we have checked the generate
+
+So please do use this ability where it helps, for example:
+
+* Add low-effort or WIP commits to fix review comments, and then squash them away before merging the PR.
+* If you have already had a PR review, don't rebase away the old commits until the PR is ready to merge, so that the reviewer only has to look at the "new" commits.
+* Rewrite the commits to make the story clearer where possible.
+
+Don't be obsessive about history though: a little bit of effort making the history clear is nice, but you can rapidly hit diminishing returns.
+Use your judgement, but probably don't merge a PR that has commits called "WIP" or "fix"!
+
+If a PR is just a total mess, consider using Github's squash-merge feature.
+
+== Pull requests
+
+All changes go through pull requests, there is no way to push to master.
+
+=== Code review and merging
+
+All pull-requests should be approved by at least one other person.
+We don't enforce this, though: a PR fixing a typo is fine to self-merge, beyond that use your judgement.
+
+As an author, code review is an opportunity for you to get feedback from clear eyes.
+As a reviewer, code review is an opportunity for you to help your colleagues and learn about what they are doing.
+Make the best use of it you can!
+
+* Pick the right reviewer(s). If you don't know who to pick, ask!
+* Respect your reviewers' time. Their time is as valuable as yours, and it's typically more efficient for you
+to spend time explaining or clarifying something in advance than for them to puzzle it out or pose a question.
+* Review code as quickly as you can. Waiting for review is often a blocker for other people, so prioritize it.
+* If you don't understand something, ask. You are as clever as any person who will read this in the future, if it confuses you it's confusing.
+* If someone had to ask about your code, it wasn't clear enough so change it or add a comment.
+* More reviewing is usually helpful. If you think a PR is interesting, you can review it even if nobody asked you to, you will probably have things to contribute and you'll learn something.
+
+=== Pull request descriptions
+
+A pull request is communication, so as usual, put yourself in the position of the reader: what does your audience (the reviewer) need to know to do their job?
+This information is easy for you to access, but hard for them to figure out, so write it down!
+
+However, better to put information in the code or commit messages if possible: these persist but PR descriptions do not.
+It's okay to repeat information from such places, or simply to point to it.
+For one-commit PRs, Github will automatically populate the PR description with the commit message, so if you've written a good commit message you're done!
+Sometimes there is "change-related" information that doesn't belong in a commit message but is useful ("Kris I think this will fix the issue you had yesterday").
```

Comment:
If you are changing UI code (could be browser but could equally be CLI), github PRs can contain GIFs so it can be really useful to add a before and after gif or an animated screenshot gif. A picture is worth a thousand words.